src/modlist.cpp

```diff
@@ -962,6 +962,40 @@ bool ModList::setActive(const QString &name, bool active)
   }
 }
 
+int ModList::setActive(const QStringList& names, bool active) {
+
+  // We only add indices for mods that exist (modIndex != UINT_MAX)
+  // and that can be enabled / disabled.
+  QList<unsigned int> indices;
+  for (const auto& name : names) {
+    auto modIndex = ModInfo::getIndex(name);
+    if (modIndex != UINT_MAX) {
+
+      // This check is not done by the bulk Profile::setModsEnabled, so we
+      // do it here.
+      ModInfo::Ptr modInfo = ModInfo::getByIndex(modIndex);
+      if (!modInfo->alwaysEnabled()) {
+        indices.append(modIndex);
+      }
+    }
+  }
+
+  if (active) {
+    m_Profile->setModsEnabled(indices, {});
+  }
+  else {
+    m_Profile->setModsEnabled({}, indices);
+  }
+
+  // Notify callbacks:
+  for (auto modIndex : indices) {
+    IModList::ModStates newState = state(modIndex);
+    m_ModStateChanged(ModInfo::getByIndex(modIndex)->name(), newState);
```

Comment:
I don't know if this is right. There should be a single call like data changed or stuff like that.
I found this: https://github.com/ModOrganizer2/modorganizer/blob/master/src/modlist.cpp#L1390

Reply:
Main concern is that you lose a lot of the advantage of doing a bulk operation if you still send off hundreds of signals that could each have a chain reaction.

Reply:
Sure, but there is no other way to do it, these are plugin-signals. I don't think those are the ones taking time unless the VFS refresh is bound on one of these, which would be very suspicious...

Reply:
I'm guessing you are testing this as you go? The does the modlist reflect changes immediately? 

On the other hand I looked at the other functions that do bulk state changes from inside Mo2 and I don't see them using m_ModStateChanged. At least not in an obvious way.

Reply:
I'm doing some testing right now, hence the WIP. The operation using the bulk method is almost instant but takes a few seconds using a for-loop + `setActive` (on a very small setup).

I am calling `m_ModStateChanged` because the non-bulk version calls it. I just want to be consistent between the two versions.